app/src/main/java/io/gnosis/safe/ui/settings/app/passcode/RepeatPasscodeFragment.kt

```diff
@@ -67,18 +71,29 @@ class RepeatPasscodeFragment : BaseViewBindingFragment<FragmentPasscodeBinding>(
                     }
                 }
                 is PasscodeViewModel.PasscodeSetup -> {
-                    if (ownerImported) {
-                        findNavController().popBackStack(R.id.ownerAddOptionsFragment, true)
+                    if (BiometricManager.from(requireContext())
+                            .canAuthenticate(BiometricManager.Authenticators.BIOMETRIC_STRONG or BiometricManager.Authenticators.BIOMETRIC_WEAK) == BiometricManager.BIOMETRIC_SUCCESS
+                    ) {
+                        val dialogBinding = DialogEnableBiometryBinding.inflate(LayoutInflater.from(context), null, false)
+                        dialogBinding.message.setText(R.string.settings_passcode_enable_biometry)
+                        CustomAlertDialogBuilder.build(
```

Comment:
also I would not show dialog automatically, but add `setup biometry` label to the screen (check legacy app) and enable biometry in on click listener

Reply:
According to the specification we're supposed to show a dialog without any additional user action. We can change that, but only with @tschubotz approval.